.github/workflows/coverage.yml

```
@@ -0,0 +1,86 @@
+name: NEURON Code Coverage
+
+on:
+  push:
+    branches:
+      - master
+      - live-debug*
+      - release/**
+  pull_request:
+    branches:
+      - master
+      - release/**
+
+jobs:
+  coverage:
+    runs-on: ubuntu-18.04
+
+    name: Code Coverage
+
+    timeout-minutes: 45
+
+    steps:
+
+      - name: Install apt packages
+        run: |
+          sudo apt-get install  build-essential doxygen autoconf automake libtool lcov autotools-dev libopenmpi-dev libmpich-dev libx11-dev libxcomposite-dev mpich openmpi-bin patchelf python-pip python-tk python-numpy python3-numpy python3-pip g++-5 g++-6 g++-8 g++-9
+        shell: bash
+
+      - name: Set up Python2
+        uses: actions/setup-python@v2
+        with:
+          python-version: '2.x'
+
+      - name: Set up Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install Python3 dependencies
+        run: python3 -m pip install --upgrade pip wheel setuptools scikit-build pytest matplotlib bokeh ipython cython pytest pytest-cov mpi4py
+
+      - uses: actions/checkout@v2
+
+      - name: Build & Test
+        id: build-test
+        shell: bash
+        working-directory: ${{runner.workspace}}/nrn
+        run:  |
+          export SHELL="/bin/bash"
+
+          # Compiler setup
+          export CC=gcc
+          export CXX=g++
+
+          # Python setup
+          export PYTHONPATH=$PYTHONPATH:$INSTALL_DIR/lib/python/
+          export PYTHON2=$(which python2);
+          export PYTHON3=$(which python3);
+
+          mkdir build && cd build;
+
+          # CMake options & flags
+          export COVERAGE_FLAGS="--coverage -O0 -fno-inline -g";
+          export CMAKE_OPTION="-DNRN_ENABLE_BINARY_SPECIAL=ON -DNRN_ENABLE_MPI=ON -DNRN_ENABLE_INTERVIEWS=ON -DNRN_ENABLE_PYTHON=ON -DNRN_ENABLE_PYTHON_DYNAMIC=ON -DNRN_PYTHON_DYNAMIC=${PYTHON2};${PYTHON3} -DNRN_ENABLE_CORENEURON=ON";
+          cmake $CMAKE_OPTION -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DNRN_ENABLE_TESTS=ON -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DCMAKE_C_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_CXX_FLAGS="${COVERAGE_FLAGS}" ..;
+
+          # Coverage
+          cmake --build . -- -j;
+          (cd ..;  lcov --capture  --initial --directory . --no-external --output-file build/coverage-base.info)
+          export PATH=`pwd`/bin:$PATH;
+          ctest -VV;
+          make install;
+          (cd ..; lcov --capture  --directory . --no-external --output-file build/coverage-run.info)
+          lcov --add-tracefile coverage-base.info --add-tracefile coverage-run.info --output-file coverage-combined.info
+          genhtml coverage-combined.info --output-directory coverage --demangle-cpp --ignore-errors source
+          bash <(curl -s https://codecov.io/bash)
```

```suggestion
          bash <(curl -s https://codecov.io/bash) -f build/coverage-combined.info
```
Otherwise it will try to rerun its own gcov and produce its own results.

---

And maybe you need to call it again to upload the python coverage reports explicitly.

---

Yeah though about that, but there are a few places for python. I plan to work on testing (specifically for code coverage purposes) and will do this at that time. For now I want to have the auto-discovery turned on.

---

But are you sure that without the -f flag it is discovering the coverage file. Remember @alkino 's struggles? in the nmodl repo the codecov bash script did never find the .info file itself and we had to force-feed it with the -f flag. When it didn't find the nicely prepared file it simply did its own thing generating coverage data and uploaded that instead.

---

Yes I'm pretty sure it's not taken into account. But better to have the code prepared for when we add `-f` and make sure lcov works in CI.

---

ahaaa.. so your point is - we have it in there but have it ignored for now. hmm.. ok :) well I don't want to be a pain about this, so if others are good with this, I don't oppose.

---

should have added a comment about that